### PR TITLE
Mapzen geocoder and routing quota check

### PR DESCRIPTION
**Pull request by ethervoid**
- [x] Added quota check for mapzen geocoder functions
- [x] Added quota check for mapzen routing functions
- [x] Refactor of the config class
- [x] Added montly_quota for mapzen services in the config

### CODE REVIEW

**Comment by ethervoid** [comment]
Please @rafatower cr this :)

File: server/extension/test/expected/00_install_test.out [hunk 1]
```diff
@@ -25,7 +25,7 @@ SELECT cartodb.cdb_conf_setconf('heremaps_conf', '{"geocoder": {"app_id": "dummy
  
 (1 row)
 
-SELECT cartodb.cdb_conf_setconf('mapzen_conf', '{"routing_app_key": "dummy_key", "geocoder_app_key": "dummy_key"}');
+SELECT cartodb.cdb_conf_setconf('mapzen_conf', '{"routing": {"api_key": "valhalla-Z61FWEs", "monthly_quota": 1500000}, "geocoder": {"api_key": "search-d744tp0", "monthly_quota": 1500000}}');
```
Comment (rafatower) [inline server/extension/test/expected/00_install_test.out]: is this a fake one, isn't it?

File: server/lib/python/cartodb_services/cartodb_services/metrics/config.py [hunk 2]
```diff
@@ -10,10 +10,15 @@ class ConfigException(Exception):
 class ServiceConfig(object):
     __metaclass__ = abc.ABCMeta
 
-    def __init__(self, redis_connection, username, orgname=None):
+    def __init__(self, redis_connection, db_conn, username, orgname=None):
         self._redis_connection = redis_connection
         self._username = username
         self._orgname = orgname
+        self._db_config = ServicesDBConfig(db_conn)
```
Comment (rafatower) [inline server/lib/python/cartodb_services/cartodb_services/metrics/config.py]: trying to figure out what this is for...

Comment (rafatower) [inline server/lib/python/cartodb_services/cartodb_services/metrics/config.py]: oh, ok! it's a reference to plpy! :)

not actually a connection in the classical sense but ok. Maybe worth a bit of doc of what's expected from that duck interface

**Comment by rafatower** [comment]
:+1:

**Pull request merged**